packages/dapper/src/transformers.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Collection of shared methods used to transform the data received from
+ * the backend to format easily consumed by the apps.
+ */
+import { PackageCardProps, SelectOption } from "@thunderstore/components";
+import { z } from "zod";
+
+import { packageCardSchema, packageCategorySchema } from "./schemas";
+
+export const transformCategories = (
+  categories: z.infer<typeof packageCategorySchema>[]
+): SelectOption[] => categories.map((c) => ({ value: c.slug, label: c.name }));
+
+export const transformPackageCards = (
+  packages: z.infer<typeof packageCardSchema>[]
+): PackageCardProps[] =>
```

Comment:
I'm not sure I fully agree. I think controlled component such as `PackageCard` should be the one that defines which props/types it accepts/requires, and encapsulate this information. The component library should be usable without other dependencies, right?

I kinda agree that this should probably cut the other way around too, i.e. Dapper shouldn't depend on the interfaces defined in `@thunderstore/components`, either. But redeclaring the interface in Dapper feels like a poor choice too. I think I'd rather keep this as is (or maybe rename the interface).

Reply:
The problem I have with that approach is that now we're tightly coupling two packages together instead of them being standalone, effectively making it so the separation makes no sense.

If we consider it from a (library) consumer perspective, it'd make sense the data provider also determines the shape of the data provided. Here it's harder to draw the lines because we're both the consumer and the provider (and the backend API) all at once, but consider a 3rd party application relying on the data provider. Now they'd be also forced to rely on our component library even if they don't use react.

I'd suggest we either reverse the dependency relation, or we define a completely separate package for common data types. I'm not sure which approach is better at this point.

So in summary, even though it intuitively makes sense for our dev process to have the dependency this way around (since the API has been designed with the client in mind), it doesn't make sense from a separation of concerns perspective & potential future library consumers.

Reply:
Well, as I stated above, IMO reversing the dependency makes no sense to me and solves nothing, so I'll guess we should look into creating a separate package. I added [this card](https://app.clickup.com/t/20413243/TS-314) for it, please check that were on the same page about this task.

Reply:
Looks good enough to me, I'll consider this solved for this PR and we can address the design later